torchdata/datapipes/iter/util/hashchecker.py

```diff
@@ -5,21 +5,25 @@
 from torchdata.datapipes import functional_datapipe
 from torchdata.datapipes.iter import IterDataPipe
 from torchdata.datapipes.utils import StreamWrapper
-from typing import Dict, Iterator, Tuple
+from typing import Dict, Iterator, Tuple, Union
+
+
+D_type = Union[str, bytes, bytearray]
+U = Union[D_type, StreamWrapper]
 
 
 @functional_datapipe("check_hash")
-class HashCheckerIterDataPipe(IterDataPipe[Tuple[str, StreamWrapper]]):
+class HashCheckerIterDataPipe(IterDataPipe[Tuple[str, U]]):
```

Comment:
In order to support both file stream and data for `HashChecker`, I have to do a `Union` here. But, the actual type should be `Union[IterDataPipe[Tuple[str, StreamWrapper]], IterDataPipe[Tuple[str, Union[str, bytes, bytearray]]]]`

cc: @VitalyFedyunin